server/schedule/checker_controller.go

```diff
@@ -34,64 +38,80 @@ type CheckerController struct {
 	ruleChecker       *checker.RuleChecker
 	mergeChecker      *checker.MergeChecker
 	jointStateChecker *checker.JointStateChecker
+	regionWaitingList cache.Cache
 }
 
 // NewCheckerController create a new CheckerController.
 // TODO: isSupportMerge should be removed.
 func NewCheckerController(ctx context.Context, cluster opt.Cluster, ruleManager *placement.RuleManager, opController *OperatorController) *CheckerController {
+	regionWaitingList := cache.NewDefaultCache(DefaultCacheSize)
 	return &CheckerController{
 		cluster:           cluster,
 		opts:              cluster.GetOpts(),
 		opController:      opController,
 		learnerChecker:    checker.NewLearnerChecker(cluster),
-		replicaChecker:    checker.NewReplicaChecker(cluster),
-		ruleChecker:       checker.NewRuleChecker(cluster, ruleManager),
+		replicaChecker:    checker.NewReplicaChecker(cluster, regionWaitingList),
+		ruleChecker:       checker.NewRuleChecker(cluster, ruleManager, regionWaitingList),
 		mergeChecker:      checker.NewMergeChecker(ctx, cluster),
```

Comment:
will we add `regionWaitingList` for `mergeChecker`?

Reply:
No, for now, we don't save the regions which need to be merged into the cache.

Reply:
So will it affect the speed of merge?

Reply:
Maybe will it be better to provide a cache for each checker